src/gnssro/gnssro_bufr2ioda.f90

```diff
@@ -223,6 +224,32 @@ program gnssro_bufr2ioda2
          nrec = nrec + 1
          ndata0 = ndata
 
+         preqc = 0
+         if (ogce .eq. 60) then
+
+            do k = 1, levs
+               if (data1b(5,k) < 1.e+9_real64 ) then
+                  if ( data1b(5,k) - roc > 25000 ) then
+                     startp = 1
+                     endp   = levs
+                     stride = 1 ! top down
+                  else
+                     startp = levs
+                     endp   = 1
+                     stride = -1 !bottom up
+                  end if
+                  exit
+              end if
+            end do
+
+            do k = startp, endp, stride
+               if ( data1b(20,k) < 1.e+9_real64 .and. data1b(18,k) <1.e+9_real64 ) then
+                  preqc(k:endp:stride) = max(int(data1b(20,k)/data1b(18,k)*100), preqc(k))
+               end if
+            end do
+
+         end if
+
```

Comment:
@hailingz so preqc is the variable containing normalized LSW value, Then we'll use the bound filter in a yaml to use this preQC, is it correct?

Reply:
Thanks! Another question: Would it be necessary to use the variable name"preQC" in this code?  Should we just use normalizedLsw (or something similar) for a more descriptive variable name. Because really here this variable is just the LSW value, not the QC yet.